service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -217,9 +217,12 @@ private String buildNotePertinentInformation(Immunization immunization) {
     private String buildExplanationPertinentInformation(Immunization immunization) {
         Optional<String> explanation;
         if (immunization.hasExplanation() && immunization.getExplanation().hasReason()) {
-            CodeableConcept reason = immunization.getExplanation().getReasonFirstRep();
-            explanation = CodeableConceptMappingUtils.extractTextOrCoding(reason);
-            return explanation.map(value -> REASON + value).orElse(StringUtils.EMPTY);
+            String reasonGiven = immunization.getExplanation().getReason().stream()
+                .map(CodeableConceptMappingUtils::extractTextOrCoding)
+                .map(Optional::get)
```

Comment:
It's not safe to directly call get on an Optional. Better to use this
```
String reasonNotGiven = immunization.getExplanation().getReason().stream()
    .map(CodeableConceptMappingUtils::extractTextOrCoding)
    .flatMap(Optional::stream)
    .filter(StringUtils::isNotBlank)
    .collect(Collectors.joining(StringUtils.SPACE));
```
I've also included `.filter(StringUtils::isNotBlank)` in the code example

Reply:
Addressed and applied to related PR: https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/320